src/librustc/hir/mod.rs

```diff
@@ -340,12 +340,16 @@ impl PathSegment {
     }
 }
 
+#[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
+pub enum PathParam {
```

Comment:
See https://github.com/rust-lang/rust/pull/48149#discussion_r167673377
>Also the things in this PR are GenericArgs, not GenericParameters (these already exist) or especially PathParameters (it's hard to come up with worse name, they are neither parameters, nor apply to a path), good opportunity to rename!

If you keep this naming, I'll rename it later anyway, but it would be nice to avoid double cruft.

Reply:
Oh, sorry, I forgot you mentioned this! What do you suggest renaming `PathParameters` to?

Reply:
`PathParam` -> `GenericArg`, `PathParameters` -> `GenericArgs`.